inference-engine/src/mkldnn_plugin/nodes/mkldnn_one_hot_node.cpp

```diff
@@ -56,32 +49,62 @@ MKLDNNOneHotNode::MKLDNNOneHotNode(const std::shared_ptr<ngraph::Node>& op, cons
     errorPrefix = "OneHot layer with name '" + op->get_friendly_name() + "'";
     const auto oneHot = std::dynamic_pointer_cast<const ngraph::opset1::OneHot>(op);
     const auto depthNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(DEPTH_ID));
-    const auto onValueNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(ON_VALUE_ID));
-    const auto offValueNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(OFF_VALUEAXES_ID));
-    depth = depthNode->cast_vector<uint32_t>()[0];
+    if (depthNode) {
+        depth = depthNode->cast_vector<uint32_t>()[0];
+    }
     axis = oneHot->get_axis();
-    src_dims = oneHot->get_input_shape(INDICES_ID);
-    if (ngraph::is_scalar(src_dims)) {
-        src_dims = SizeVector{1};
+
+    VectorDims srcDims = getInputShapeAtPort(INDICES_ID).getDims();
+    if (ngraph::is_scalar(srcDims)) {
+        srcDims = SizeVector{1};
     }
-    dst_dims = oneHot->get_output_shape(0);
-    if (ngraph::is_scalar(dst_dims)) {
-        dst_dims = SizeVector{1};
+    VectorDims dstDims = getOutputShapeAtPort(0).getDims();
+    if (ngraph::is_scalar(dstDims)) {
+        dstDims = SizeVector{1};
     }
 
-    int output_dims_size = dst_dims.size();
+    int output_dims_size = dstDims.size();
     if (axis < 0) {
         axis += output_dims_size;
     }
     if (axis < 0 || axis >= output_dims_size) {
         IE_THROW() << errorPrefix << " has unsupported 'axis' attribute: " << oneHot->get_axis();
     }
 
-    if (!( ((1 + src_dims.size()) == dst_dims.size()) ||
-           (src_dims.size() == 1 && dst_dims.size() == 1 && dst_dims[0] == depth && src_dims[0] == 1)))
+    if (!(((1 + srcDims.size()) == dstDims.size()) ||
+            (depthNode && (srcDims.size() == 1 && dstDims.size() == 1 && dstDims[0] == depth && srcDims[0] == 1))))
         IE_THROW() << errorPrefix << " has incorrect number of input/output dimensions!";
 }
 
+bool MKLDNNOneHotNode::needShapeInfer() const {
+    const auto depthNodePtr = reinterpret_cast<int32_t *>(getParentEdgesAtPort(1)[0]->getMemoryPtr()->GetPtr());
+    if (depth != depthNodePtr[0])
+        return true;
+    return MKLDNNNode::needShapeInfer();
+}
+
+std::vector<VectorDims> MKLDNNOneHotNode::shapeInfer() const {
+    std::vector<ov::StaticShape> input_shapes = {
```

Comment:
That's definitely to much overhead for oneHot shape interference routine given simplicity of its math. We need to discuss how this can be optimized. At least some initializations can be moved on compilation stage. 
cc'ed @mandrono  @maxnick

Reply:
@egorshul 
looks like for OneHot enough copy input dims and insert depth value on axis position
Please fix `shapeInfer()`

Reply:
PR with optimization [#8739](https://github.com/openvinotoolkit/openvino/pull/8739)